File: 2026-04/src/day_10/boj_1292/Main.java
package day_10.boj_1292;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int[] arr = new int[1001];
        int num = 1;
        int idx = 1;

        out:
        while (true) {
            for (int i = 1; i <= num; i++) {
                arr[idx++] = num;
                if (idx > 1000) break out;
            }
            num++;
        }

        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());

        int sum = 0;
        for (int i = a; i <= b; i++) {
            sum += arr[i];
        }

        System.out.println(sum);
    }
}

File: 2026-04/src/day_10/boj_1292/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int arr[1001];

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int num = 1;
    int idx = 1;
    bool flag = true;

    while (flag) {
        for (int i = 1; i <= num; i++) {
            arr[idx++] = num;
            if (idx > 1000) {
                flag = false;
                break;
            }
        }
        num++;
    }

    int a, b;
    cin >> a >> b;

    int sum = 0;
    for (int i = a; i <= b; i++) {
        sum += arr[i];
    }

    cout << sum;
}
